### PR TITLE
Add js library as option for webpack config

### PR DESCRIPTION
**Pull request by Mivr**
### Changes are visible to end-users: yes

- Searched for relevant documentation and updated as needed: YES
- Breaking change (forces users to change their own code or config): NO
- Suggested release notes appear below: YES

Added option to use js_library for webpack config as alternative to files.

### Test plan

- New test cases added

### CODE REVIEW

File: webpack/private/webpack_bundle.bzl [hunk 1]
```diff
@@ -117,7 +118,7 @@ def _impl(ctx):
     for (key, value) in ctx.attr.env.items():
         env[key] = " ".join([
             expand_variables(ctx, exp, attribute_name = "env")
-            for exp in expand_locations(ctx, value, entry_points_srcs + ctx.attr.srcs + ctx.attr.deps + ctx.attr.data).split(" ")
+            for exp in expand_locations(ctx, value, entry_points_srcs + ctx.attr.srcs + ctx.attr.deps + ctx.attr.data + ctx.attr.webpack_configs).split(" ")
```
Comment (jbedard) [inline webpack/private/webpack_bundle.bzl:121]: Why was this one required?

File: e2e/loaders_jslib/package.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+  "scripts": {
+    "clean": "rm -rf dist node_modules"
```
Comment (jbedard) [inline e2e/loaders_jslib/package.json:3]: Delete?

File: e2e/loaders_jslib/WORKSPACE [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+# brought as a dep by webpack ruleset
+load("@aspect_rules_js//npm:repositories.bzl", "npm_translate_lock")
+
+npm_translate_lock(
+    name = "npm",
+    no_optional = True,
+    npmrc = "//:.npmrc",
+    pnpm_lock = "//:pnpm-lock.yaml",
+    pnpm_version = "6.29.1",
```
Comment (jbedard) [inline e2e/loaders_jslib/WORKSPACE:27]: Can we upgrade this to v9? I hope to remove at least v6+7 in the next rules_js major release

File: webpack/private/webpack_bundle.bzl (hunk 1 above)
Comment (Mivr) [reply webpack/private/webpack_bundle.bzl:121]: Its not, removed

File: e2e/loaders_jslib/package.json (hunk 2 above)
Comment (Mivr) [reply e2e/loaders_jslib/package.json:3]: Removed the scripts

File: e2e/loaders_jslib/WORKSPACE (hunk 3 above)
Comment (Mivr) [reply e2e/loaders_jslib/WORKSPACE]: Updated

**Review by jbedard — APPROVED** [review]

**Pull request merged**